src/Mailgun/Connection/RestClient.php

```diff
@@ -25,30 +25,59 @@ class RestClient
     private $apiKey;
 
     /**
-     * @var Guzzle
+     * @var HttpAdapter
      */
-    protected $mgClient;
+    protected $httpAdapter;
+
+    /**
+     * @var string
+     */
+    protected $apiEndpoint;
 
     /**
      * @param string $apiKey
-     * @param string $apiEndpoint
+     * @param string $apiHost
      * @param string $apiVersion
      * @param bool   $ssl
      */
-    public function __construct($apiKey, $apiEndpoint, $apiVersion, $ssl)
+    public function __construct($apiKey, $apiHost, $apiVersion, $ssl, HttpAdapter $adapter = null)
     {
         $this->apiKey = $apiKey;
-        $this->mgClient = new Guzzle([
-            'base_url'=>$this->generateEndpoint($apiEndpoint, $apiVersion, $ssl),
-            'defaults'=>[
-                'auth' => array(Api::API_USER, $this->apiKey),
-                'exceptions' => false,
-                'config' => ['curl' => [ CURLOPT_FORBID_REUSE => true ]],
```

Comment:
Thank you for your input. Do you know who are on the Mailgun team that can confirm this?